mh/pat.py

```diff
@@ -2412,15 +2439,25 @@ def handle(self):
         self.server.info("Handle client")
         self.server.add_to_debug(self)
         self.session = Session(self)
+        exception_raised = False
 
+        # Temporary workaround for Dolphin bug
+        time.sleep(2)
         self.sendReqConnection()
         try:
             self.handle_client()
-            self.session.disconnect()
         except Exception as e:
+            exception_raised = True
             self.server.error(traceback.format_exc())
             self.send_error("{}: {}".format(type(e).__name__, str(e)))
-            self.session.delete()
+
+        self.session.disconnect()  # Always disassociate capcom_id with current session
+        if not self.finished or exception_raised:  # Abnormal disconnection, clean up
+            try:
+                self.notify_layer_departure()
```

Comment:
It's missing an `except Exception as e:` block. Without it, the rest of the code after the `finally` block won't be executed.